articles/kotlin-basics/index.md

```diff
@@ -0,0 +1,1605 @@
+- Or refer to **repl.it** [click here](https://repl.it/join/uovhpspi-stephenchacha).
+
+### Topics to be covered
+1. Data Types and Variables
+2. Constants, Variables, and Data Types
+3. Control Flow Statements
+4. Loop Control Statements
+5. Functions and Interoperability
+6. Object Oriented Programming in Kotlin
+7. Functional Programming in Kotlin
+8. Kotlin NULL Safety
+ 
+```kotlin
+fun main(args: Array<String>) {
+    print("Hello World")
+}
+
+// Output: Hello world
+```
+
+### 1. Defining Variables
+
+#### `val`
+It can be assigned a value only once 
+They are immutable string
+
+``` kotlin
+fun main(args: Array<String>) {
+    val x = 2
+    println("value x is : $x")
+}
+
+// Output: 2
+```
+
+ #### `var`
+It can be reassigned a value after assigned locally
+They are mutable string
+
+```kotlin
+fun main() {
+    var y = 10
+    y += 2
+    println(y)
+}
+
+// Output: 12
+```
+#### Comments
+Like in other programming languages comments are used to explain or highlight what you are talking about. It helps in the documentation for developers.
+Comments are ignored during code compilation
+
+Comments can be written in two ways:
+
+* Using block comments
+```kotlin
+/*
+* This is comment line 1
+* */
+```
+
+* By use of double forward slashes
+```kotlin
+fun main (){
+    // This is a comment
+}
+```
+
+### Data types
+In kotlin data, types are all objects.
+They include:
+
+| Data type         | size          |
+|-------------------|:-------------:|
+| Boolean           | 1 bit         |
+| Byte              | 8 bit         |
+| Char              | 16 bit        |
+| Short             | 16 bit        |
+| Int               | 32 bit        |
+| Long              | 64 bit        |
+| Float             | 32 bit        |
+| Double            | 64 bit        |
+
+#### Numbers
+Kotlin provides a set of built-in types that represent numbers.
+For integer numbers, there are four types with different sizes and, hence, value ranges.
+
+```kotlin
+fun main() {
+    val num1= 1 //Int
+    val One_Billion=1000000000 //Long
+    val num1 = 53L // Long
+    val num2 = 3.14 // Double  
+    var marks: Float = 287.4F
+
+    var age: Int = 10 // immidiate assignment
+    var myAge = 10   // int type is inferred
+    
+    println(num1)
+    //1
+    println(num2)
+    //3.14
+
+    println(myAge)
+    // 10
+    println(marks)
+    // 287.4
+}
+```
+
+
+#### Characters
+Characters are represented by the use of the keyword ``Char```. They can not be treated directly as numbers
+
+```kotlin
+
+fun main() {
+
+    var gender: Char = 'M'
+}
+```
+
+#### Boolean
+Kotlin also contains Boolean types. It has two values of ``false`` or ``true``
+
+```kotlin
+
+fun main() {
+    
+    var isAlive: Boolean = true
+}
+```
+ #### String
+
+```kotlin
+
+fun main() {
+
+    var name: String
+    name = "Steve"
+    printlin(name)
+}
+
+// Output: Steve
+```
+
+#### Arrays
```

Comment:
This is a repetition of [this article](https://www.section.io/engineering-education/kotlin-collections/) you could just refer to it instead of writing all over again